src/sentry/seer/similarity/utils.py

```diff
@@ -446,3 +454,89 @@ def set_default_project_seer_scanner_automation(
         project.update_option(
             "sentry:default_seer_scanner_automation", org_default_seer_scanner_automation
         )
+
+
+def report_token_count_metric(
+    event: Event | GroupEvent,
+    variants: dict[str, BaseVariant] | None,
+    outcome: str,
+) -> None:
+    """
+    Calculate token count and report metrics for stacktrace token analysis.
+
+    This function is gated by the 'seer.similarity.token_count_metrics_enabled' option
+    and will do nothing if disabled.
+
+    Args:
+        event: A Sentry Event object containing stack trace data
+        variants: Optional pre-calculated grouping variants to avoid recalculation
+        outcome: The frame check outcome ("pass", "block", "bypass")
+    """
+    if not options.get("seer.similarity.token_count_metrics_enabled", False):
+        return
+
+    platform = event.platform or "unknown"
+
+    token_count = get_token_count(event, variants, platform)
+
+    metrics.distribution(
+        "grouping.similarity.token_count",
+        token_count,
+        sample_rate=options.get("seer.similarity.metrics_sample_rate"),
+        tags={
+            "platform": platform,
+            "frame_check_outcome": outcome,
+        },
+    )
+
+
+def get_token_count(
+    event: Event | GroupEvent, variants: dict[str, BaseVariant], platform: str
+) -> int:
+    """
+    Count the number of tokens in the stack trace of an event.
+
+    Stacktrace string should be already cached on the event, and only calculates it if needed.
+
+    Args:
+        event: A Sentry Event object containing stack trace data
+        variants: Pre-calculated grouping variants to avoid recalculation
+
+    Returns:
+        The number of tokens in the stack trace text
+    """
+    with metrics.timer(
+        "grouping.similarity.get_token_count",
+        sample_rate=options.get("seer.similarity.metrics_sample_rate"),
+        tags={"platform": platform},
+    ) as timer_tags:
+        try:
+            timer_tags["has_content"] = False
+            timer_tags["cached"] = event.data.get("stacktrace_string") is not None
+            timer_tags["source"] = "stacktrace_string"
```

Comment:
```suggestion
            timer_tags["source"] = "cached_string"
```
(To better differentiate from the `get_stacktrace_string` source.)

Reply:
but there's a "cached" tag, though I guess I can just lose that and unifty it

Reply:
actually I prefer it like this, I dont want to have to write "non_cached_string" on the other, and I dont want it to just say "stacktrace_string" and someone to need to know "cached_string" also exists.